pkg/egressgateway/policy_test.go

```diff
@@ -0,0 +1,181 @@
+							MatchLabels: map[string]string{
+								"node-name": "node1",
+							},
+						},
+					},
+				},
+			},
+			args: args{
+				epDataStore: map[endpointID]*endpointMetadata{
+					"123456": {
+						labels: map[string]string{
+							"app": "test",
+						},
+						nodeIP: "192.168.1.10",
+					},
+				},
+				nodesAddresses2Labels: map[string]map[string]string{
+					"192.168.1.10": {
+						"node-name": "node1",
+					},
+				},
+			},
+			want: 1,
+		},
+		{
+			name: "Test updateMatchedEndpointIDs endpoints and nodes with no match",
+			fields: fields{
+				id: types.NamespacedName{
+					Name: "test",
+				},
+				endpointSelectors: []api.EndpointSelector{
+					{
+						LabelSelector: &slimv1.LabelSelector{
+							MatchLabels: map[string]string{
+								"app": "test",
+							},
+						},
+					},
+				},
+				nodeSelectors: []api.EndpointSelector{
+					{
+						LabelSelector: &slimv1.LabelSelector{
+							MatchLabels: map[string]string{
+								"node-name": "node1",
+							},
+						},
+					},
+				},
+			},
+			args: args{
+				epDataStore: map[endpointID]*endpointMetadata{
+					"123456": {
+						labels: map[string]string{
+							"app": "test",
+						},
+						nodeIP: "192.168.1.10",
+					},
+				},
+				nodesAddresses2Labels: map[string]map[string]string{
+					"192.168.1.11": {
+						"node-name": "node1",
+					},
+				},
+			},
+			want: 0,
+		},
+		{
+			name: "Test updateMatchedEndpointIDs endpoints and nodes with no match label",
+			fields: fields{
+				id: types.NamespacedName{
+					Name: "test",
+				},
+				endpointSelectors: []api.EndpointSelector{
+					{
+						LabelSelector: &slimv1.LabelSelector{
+							MatchLabels: map[string]string{
+								"app": "test",
+							},
+						},
+					},
+				},
+				nodeSelectors: []api.EndpointSelector{
+					{
+						LabelSelector: &slimv1.LabelSelector{
+							MatchLabels: map[string]string{
+								"node-name": "node1",
+							},
+						},
+					},
+				},
+			},
+			args: args{
+				epDataStore: map[endpointID]*endpointMetadata{
+					"123456": {
+						labels: map[string]string{
+							"app": "test",
+						},
+						nodeIP: "192.168.1.10",
+					},
+				},
+				nodesAddresses2Labels: map[string]map[string]string{
+					"192.168.1.10": {
+						"bar": "bar",
+					},
+				},
+			},
+			want: 0,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := &PolicyConfig{
+				id:                tt.fields.id,
+				endpointSelectors: tt.fields.endpointSelectors,
+				nodeSelectors:     tt.fields.nodeSelectors,
+				dstCIDRs:          tt.fields.dstCIDRs,
+				excludedCIDRs:     tt.fields.excludedCIDRs,
+				policyGwConfig:    tt.fields.policyGwConfig,
+				matchedEndpoints:  tt.fields.matchedEndpoints,
+				gatewayConfig:     tt.fields.gatewayConfig,
+			}
+			config.updateMatchedEndpointIDs(tt.args.epDataStore, tt.args.nodesAddresses2Labels)
+			assert.Len(t, config.matchedEndpoints, tt.want)
```

Comment:
small thing - should we make this assertion a bit more specific than just the count of matched pods?

Reply:
Done.